lib/datadog/tracing/contrib/http/distributed/fetcher.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+# typed: false
+
+require_relative '../../../distributed/fetcher'
+require_relative '../../../../core/utils/refinement'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module HTTP
+        module Distributed
+          # Retrieves Rack formatted headers from HTTP headers.
+          class Fetcher < Tracing::Distributed::Fetcher
+            using Core::Utils::Refinement::String unless ::String.method_defined?(:+@)
+
+            # TODO: Don't assume Rack format.
+            #       Make distributed tracing headers apathetic.
+            # DEV: Should we try to parse both verbatim an Rack-formatted headers,
+            # DEV: given Rack-formatted is the most common format in Ruby?
+            def [](name)
+              # DEV: `String#+@` is not needed in Ruby >= 3.0, as interpolated strings are not frozen.
+              rack_header = (+"http-#{name}").upcase!.tr('-', '_')
```

Comment:
> Furthermore, as this is an optimization, should we use tr! instead of tr?

I missed this one, thank you!

Reply:
I find that this pattern of "is this string mutable, if not dup" is common in other parts of the tracer, and `+@` is super helpful. Because I'm removing the use case here, I'll remove the refinement, but I think it's a great tool to have in other scenarios.

Reply:
Removed refinement changes and made suggested changes.

Reply:
> I find that this pattern of "is this string mutable, if not dup" is common in other parts of the tracer, and +@ is super helpful. Because I'm removing the use case here, I'll remove the refinement, but I think it's a great tool to have in other scenarios.

I guess it's just me not being used to either `+` or refinements, so I may be putting the bar to use them a bit higher, e.g. I think that indeed we may end up using them, but I think in this case the added complexity was not completely worth it.